fastestimator/op/tensorop/augmentation/mixup_batch.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Any, Dict, List, TypeVar, Union
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+import torch
+
+from fastestimator.op.tensorop import TensorOp
+
+Tensor = TypeVar('Tensor', tf.Tensor, torch.Tensor)
+
+
+class MixUpBatch(TensorOp):
+    """ This class should be used in conjunction with MixUpLoss to perform mix-up training, which helps to reduce
+    over-fitting, stabilize GAN training, and against adversarial attacks (https://arxiv.org/abs/1710.09412)
+
+    Args:
+        inputs: key of the input to be mixed up
+        outputs: key to store the mixed-up input
+        mode: what mode to execute in. Probably 'train'
+        alpha: the alpha value defining the beta distribution to be drawn from during training
+        sharedbeta: Sample a single beta for a batch Or single beta for each image
+    """
+    def __init__(self, inputs: Union[str, List[str]] = None,
+                 outputs: Union[str, List[str]] = None,
```

Comment:
outputs typing should only be List[str] since need both a lambda key and an output key. no reason 1 string would ever be acceptable. We should have an assert to check this in the init